src/fetchers/tarball-fetcher.js

```diff
@@ -208,11 +216,7 @@ export default class TarballFetcher extends BaseFetcher {
       return this.fetchFromLocal(this.reference);
     }
 
-    if (await this.getLocalAvailabilityStatus()) {
-      return this.fetchFromLocal();
-    } else {
-      return this.fetchFromExternal();
-    }
+    return this.fetchFromLocal().catch(err => this.fetchFromExternal());
```

Comment:
This will apply the fallback for any error - we probably only want to do this for a small subset of errors (basically ENOENT)